src/vcpkg/binarycaching.cpp

```diff
@@ -1007,13 +1033,10 @@ namespace
             {
                 auto&& action = *actions[idx];
                 const auto& abi = action.package_abi().value_or_exit(VCPKG_LINE_INFO);
-                if (m_tool->stat(make_object_path(m_prefix, abi)))
-                {
-                    cache_status[idx] = CacheAvailability::available;
-                }
-                else
+                auto maybe_res = m_tool->stat(make_object_path(m_prefix, abi));
+                if (auto res = maybe_res.get())
                 {
-                    cache_status[idx] = CacheAvailability::unavailable;
+                    cache_status[idx] = *res;
                 }
```

Comment:
Just realized that this is not an equivalent transformation: if `stat()` returns an error, the original version sets `cache_status[idx]` to `CacheAvailability::unavailable`, while the new version leaves it as `CacheAvailability::unknown`. Should I add an `else` branch to keep the behaviour the same as before? (My understanding is that leaving it `unknown` may at worst cause an extra query to the provider, but I'm not 100% sure I'm reading the rest of the code right.)

Reply:
GitHub's not showing the diff properly.

Original:
```
                if (m_tool->stat(make_object_path(m_prefix, abi)))
                {
                    cache_status[idx] = CacheAvailability::available;
                }
                else
                {
                    cache_status[idx] = CacheAvailability::unavailable;
                }
```

Current:
```
                auto maybe_res = m_tool->stat(make_object_path(m_prefix, abi));
                if (auto res = maybe_res.get())
                {
                    cache_status[idx] = *res;
                }
```

Possible fix to keep old behaviour:
```
                auto maybe_res = m_tool->stat(make_object_path(m_prefix, abi));
                if (auto res = maybe_res.get())
                {
                    cache_status[idx] = *res;
                }
                else
                {
                    cache_status[idx] = CacheAvailability::unavailable;
                }
```

Reply:
I decided to go with the real equivalent change (i.e. I readded the `else` branch), and force-pushed.